spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestMetadataTables.java

```diff
@@ -671,6 +671,76 @@ public void testFilesTableTimeTravelWithSchemaEvolution() throws Exception {
         .hasSameSizeAs(expectedFiles);
   }
 
+  @TestTemplate
+  public void testFilesTableVersionAsOfTag() {
```

Comment:
Looks like this test will pass even without the change in this PR. But the next AsOfBranch will fail

Reply:
Yes, do you want to remove this test for separate a PR?

Reply:
I think we can keep them